libs/SalesforceSDKCore/SalesforceSDKCore/Classes/PushNotification/PushNotificationManager.swift

```diff
@@ -79,45 +79,31 @@ public class PushNotificationManager: NSObject {
     
     var isSimulator: Bool = false
     private let notificationRegister: RemoteNotificationRegistering
-    private let apiVersion: String
-    private let restClient: RestClient?
-    private let currentUser: UserAccount?
-    private let preferences: SFPreferences?
+    private var preferences: SFPreferences?
+    private var loginObserver: NSObjectProtocol?
+    private var enterForegroundObserver: NSObjectProtocol?
     
     /// Convenience initializer that sets up the PushNotificationManager with default values:
     /// - notificationRegister: DefaultRemoteNotificationRegistrar() - Handles APNS registration
-    /// - apiVersion: RestClient.shared.apiVersion - Uses the current API version from RestClient
     /// - restClient: RestClient.shared - Uses the shared RestClient instance
-    /// - currentUser: UserAccountManager.shared.currentUserAccount - Uses the current logged-in user
     /// - preferences: SFPreferences.sharedPreferences(for: .user, user: currentUser) - Uses user-level preferences
     @objc
     public override convenience init() {
         self.init(notificationRegister: DefaultRemoteNotificationRegistrar(),
-                   apiVersion: RestClient.shared.apiVersion,
-                   restClient: RestClient.shared,
-                   currentUser: UserAccountManager.shared.currentUserAccount,
-                   preferences: SFPreferences.sharedPreferences(for: .user, user: UserAccountManager.shared.currentUserAccount))
+                  preferences: SFPreferences.currentUserLevel())
```

Comment:
I think the user preferences could get out of sync on user logout / login / switch if it's set at init, is it possible to not store it at the class level like you did with the user account and rest client?

Reply:
Yes.  I'll do a similar pattern for preferences